packages/flutter/test/widgets/container_test.dart

```diff
@@ -586,6 +586,52 @@ void main() {
     expect(tapped, false);
   });
 
+    testWidgets('Container transformAlignment', (WidgetTester tester) async {
+    bool didReceiveTap = false;
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Stack(
+          children: <Widget>[
+            Positioned(
+              top: 100.0,
+              left: 100.0,
+              child: Container(
+                width: 100.0,
+                height: 100.0,
+                color: const Color(0xFF0000FF),
+              ),
+            ),
+            Positioned(
+              top: 100.0,
+              left: 100.0,
+              child: Container(
+                width: 100.0,
+                height: 100.0,
+                transform: Matrix4.diagonal3Values(0.5, 0.5, 1.0),
+                transformAlignment: const Alignment(1.0, 0.0),
+                child: GestureDetector(
+                  onTap: () {
+                    didReceiveTap = true;
+                  },
+                  child: Container(
+                    color: const Color(0xFF00FFFF),
+                  ),
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+
+    expect(didReceiveTap, isFalse);
+    await tester.tapAt(const Offset(110.0, 110.0));
```

Comment:
Instead of testing this indirectly via hit test, can we just check that the corners of the transformed widget are aat the expected location?

Reply:
Hi @goderbauer I am not quite sure how I can do that. I have done rest of the changes

Reply:
`renderObject.localToGlobal(Offset.zero)` should work, other tests do something similar.

Reply:
Hi idk what happened suddenly a lot of commits popped up?